tests/PHPStan/Type/TypeCombinatorTest.php

```diff
@@ -531,7 +531,7 @@ public function dataUnion(): array
 					new CallableType(),
 				],
 				UnionType::class,
-				'callable|RecursionCallable\Foo',
+				'callable(): mixed|RecursionCallable\Foo',
```

Comment:
This looks ambiguous. Is it `(callable(): mixed)|RecursionCallable\Foo` or `callable(): (mixed|RecursionCallable\Foo)`?

Reply:
I agree that it looks ambiguous. I made some changes.